ext/git.c

```diff
@@ -198,27 +202,30 @@ void ddtrace_inject_git_metadata(zval *git_metadata_zv) {
         cwd_zstr = zend_string_init(cwd, strlen(cwd), 1);
     }
 
-    zval *entry = zend_hash_find(&DDTRACE_G(git_metadata), cwd_zstr);
-
-    if (entry && Z_TYPE_P(entry) == IS_FALSE) {
-        zend_string_release(cwd_zstr);
-        return;
-    }
+    struct _git_metadata *entry = zend_hash_find_ptr(&DDTRACE_G(git_metadata), cwd_zstr);
 
     if (entry) {
-        zend_string_release(cwd_zstr);
-        ZVAL_COPY(git_metadata_zv, entry);
-        return;
-    }
-
-    if (inject_from_env(git_metadata_zv) || inject_from_global_tags(git_metadata_zv) || inject_from_git_dir(git_metadata_zv, cwd_zstr)) {
-        zend_hash_add(&DDTRACE_G(git_metadata), cwd_zstr, git_metadata_zv);
+        if (entry->available) {
+            add_git_info(git_metadata_zv, entry->property_commit, entry->property_repository);
+        }
+    } else if (inject_from_env(git_metadata_zv) || inject_from_global_tags(git_metadata_zv) || inject_from_git_dir(git_metadata_zv, cwd_zstr)) {
+        struct _git_metadata *git_metadata = pemalloc(sizeof(struct _git_metadata), 1);
+        git_metadata->available = true;
+        zend_string *property_commit = Z_STR_P(&((ddtrace_git_metadata *) Z_OBJ_P(git_metadata_zv))->property_commit);
+        zend_string *property_repository = Z_STR_P(&((ddtrace_git_metadata *) Z_OBJ_P(git_metadata_zv))->property_repository);
```

Comment:
```suggestion
        zend_string *property_commit = Z_STR(((ddtrace_git_metadata *) Z_OBJ_P(git_metadata_zv))->property_commit);
        zend_string *property_repository = Z_STR(((ddtrace_git_metadata *) Z_OBJ_P(git_metadata_zv))->property_repository);
```
These all have equivalents without _P too, saving the dereference.